pkg/pillar/scripts/onboot.sh

```diff
@@ -115,32 +115,74 @@ if [ ! -d $PERSISTDIR/status ]; then
     mkdir $PERSISTDIR/status
 fi
 
-# Checking for low diskspace at bootup. If used percentage of
-# /persist directory is more than 70% then we will remove the
+# percent_used <dataset name> (without leading '/')
+percent_used() {
+    res=$(zfs list -pH -o available,used "$1")
+    # shellcheck disable=SC2181
+    if [ $? = 0 ]; then
+        # shellcheck disable=SC2086
+        avail=$(echo $res | cut -d\  -f1)
+        # shellcheck disable=SC2086
+        used=$(echo $res | cut -d\  -f2)
+        echo $((100*used/(avail+used)))
+    else
+        df --sync /"$1" |awk '{printf("%d",$5);}'
+    fi
+}
+
+# free_space <dataset name> (without leading '/')
+# return value is truncated to MBytes
+# Note that we use df even for zfs since the "available" property in zfs
+# includes unused space in child datasets
+free_space() {
+    ds="$1"
+    res=$(df --sync -k --output=avail "/$ds" | tail -1)
+    echo $((res/1024))
+}
+
+# Checking for low diskspace at bootup.
+# If there is less than 4Mbytes (MIN_DISKSPACE) then remove the content of the
+# following directories in order until we have that amount of available space
 # following sub directories:
 # /persist/log/*
-# /persist/newlog/appUpload/*
-# /persist/newlog/devUpload/*
+# /persist/pubsub-large/*
+# /persist/netdump
 # /persist/newlog/keepSentQueue/*
 # /persist/newlog/failedUpload/*
-diskspace_used=$(df /persist |awk '/\/dev\//{printf("%d",$5);}')
-echo "Used percentage of /persist: $diskspace_used"
-if [ "$diskspace_used" -ge "$DISKSPACE_RECOVERY_LIMIT" ]
+# /persist/newlog/appUpload/*
+# /persist/newlog/devUpload/*
+# /persist/containerd-system-root
+# /persist/vault/containerd - NO causes failures to start containers
+# /persist/vault/downloader
+# /persist/vault/verifier
+# /persist/agentdbug
```

Comment:
I would also add the files created by `collect-info.sh`. They definitely take up space.
Also, `/persist/memory-monitor/output/*`: can free up to 100 Mb.
Also, the kernel dump files. @rouming, could you please remind us where we store them? And is there a filename pattern that we can use for this code?

Reply:
Suggestions for relative priority among those? We delete in priority order in the loop below and then check whether we got sufficient space, so adding things to the end means that are more likely to be preserved.

Reply:
I've just checked the collect-info script and see the pattern for the collect-info archives is:
`/persist/eve-info-v<version>-<date>.tar.gz`. It may make sense to fix the script so the archives appear in some dedicated directory so we can add it to the list here. The other option is adding some logic that removes the most old archives. I would remove them first, as users usually copy the archives immediately after creation. 

Regarding the kernel dumps: according to the documentation (https://github.com/lf-edge/eve/blob/ca5768e44c5f4a9966d2d57995bc753d6fb54efb/docs/KERNEL-DUMPS.md#L50), they will be created in `/persist/kcrashes`. So we can add it here right now. 

Regarding the priorities, I would remove the debug data first, as it does not affect the system's functionality — it will not force rerun downloading, re-verification of the images, etc. But maybe someone has another opinion.

Reply:
OK, I'm adding those. Should we do a separate PR to move eve-info into a subdirectory of persist?